RHOAIENG-27684: The workbench image column shows Deleted label after RHOAI upgrade

srtanish1992 opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-123456 -->
**JIRA:** https://issues.redhat.com/browse/RHOAIENG-27684

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->

I have fixed the logic to check if the notebook image has been deleted and added additional logic to check if the image has been updated/outdated.

https://github.com/user-attachments/assets/d7731fba-1a45-4419-80ef-3e00e7d7fd52

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

1. Install the RHOAI 2.21
2. Create a DS project and create a Standard Data Science workbench there with version 2025.1
3. Perform upgrade to RHOAI 2.22
4. You should see `Deprecated` instead of the `Deleted` label in the RHOAI Dashboard for the running workbench.

## Test Impact
<!--- What tests have you done to cover the implemented functionality -->
<!--- If tests are not applicable, explain why here -->
Fixed tests and updated deleted, deprecated test.

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Clearer notebook image status labels — shows “Deprecated” for commit/tag mismatches and “Deleted” when the selected tag is missing; updated displayed names.

* **Bug Fixes**
  * More accurate deprecation/deletion detection by honoring the notebook’s last image selection (name:tag) and BYON-aware checks across all resolution paths.
  * Consistent image status evaluation so detail and workbench views reflect tag/commit state.

* **Tests**
  * Updated unit and E2E mocks/assertions to use explicit image tag/version selections and cover deprecated vs deleted scenarios.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

srtanish1992 commented:
/test-e2e

Gkrumbach07 reviewed (changes requested):
Each one of `getNotebookImageInternalRegistry`, `getNotebookImageInternalRegistry`, and `getNotebookImageNoInternalRegistryNoSHA` all call isNotebookImageDeleted(notebook, imageStream). in that func we get the image tag by 

```tsx
const [lastImageSelectionName, lastImageSelectionTag] =
    notebook.metadata.annotations?.['notebooks.opendatahub.io/last-image-selection']?.split(':') ??
    [];
```

but in reality we should be passing in either `versionName ` or `lastImageSelectionTag` from the parent func

srtanish1992 commented:
> Each one of `getNotebookImageInternalRegistry`, `getNotebookImageInternalRegistry`, and `getNotebookImageNoInternalRegistryNoSHA` all call isNotebookImageDeleted(notebook, imageStream). in that func we get the image tag by
> 
> ```tsx
> const [lastImageSelectionName, lastImageSelectionTag] =
>     notebook.metadata.annotations?.['notebooks.opendatahub.io/last-image-selection']?.split(':') ??
>     [];
> ```
> 
> but in reality we should be passing in either `versionName ` or `lastImageSelectionTag` from the parent func

I have fixed it in https://github.com/opendatahub-io/odh-dashboard/pull/4988/commits/993451d57705b96c0f0b2b30897d82127480b874. Let me know if this is what you were suggesting.

srtanish1992 commented:
> E2E Tests [#209](https://jenkins-csb-rhods-opendatascience.dno.corp.redhat.com/job/cypress/job/cypress-tests/209/) finished with result: **[UNSTABLE](https://jenkins-csb-rhods-opendatascience.dno.corp.redhat.com/job/cypress/job/cypress-tests/209/display/redirect?page=tests)**

there is only one test failure for pipelines which is unrelated to the code changes in this PR.
<img width="1614" height="109" alt="Screenshot 2025-09-25 at 1 26 17 PM" src="https://github.com/user-attachments/assets/283b5b5b-eda0-4967-80c3-5c600fb6c334" />
